docs/CONNECTOR.md

```diff
@@ -0,0 +1,51 @@
+# Connector toolkit
+This library includes a base toolkit to create a ship - connector to integrate Hull platform with a selected 3rd party API Service.
+
+## new Hull.App()
+```js
+import Hull from "hull";
+
+const app = new Hull.App({ port, hostSecret, service });
```

Comment:
@unity just to sum up our discussion here - I all agree with your arguments, I find the express based architecture a good choice too.

I will just tell you that all that parts going far from plain express app, are caused by slightly different approach which I had in mind. After discussions with you and @sbellity trying to bring together your perspectives I came to a conclusion that the express application is only some part of our ship application, btw. the part which could be replaced or changed in some point in the future (for other http framework as koa or dropped for other solution).

But after seeing all the discussion here, I clearly see that you are all for having the `express` as a first citizen of the whole architecture. So this is my proposition which should reconcile everybody here :)

1. Let's make the sacred `const app = express();` line being called within the ship
2. Let's make the current `Hull.App` a class returning a set of middlewares, routers and commands (so a method which takes the `app` and mutates it, applying some custom logic) - my perspective here is if I were the external ship developer I would find the `app = express` approach most comfortable for the reason you've brought here up. Yet observing the first actual developers doing their first ships I see that they are much much more focused on the business logic, the parts related to the 3rd party API that to the ships internals or infrastructure parts. I would say that the mind flow of a developer could be totally opposite to the natural flow of the application. That's why I will suggest having a "super command" method where you can start very easily having all the basics set up on our current convention (like manifest, README, templates), then you can turn it off and switch to more low level elements.